ci/ci/build.py

```diff
@@ -89,8 +89,9 @@ def __init__(self, code, config_str, scope, requested_step_names=None):
         for step_config in config['steps']:
             step_params = StepParameters(code, scope, step_config, name_step)
             step = Step.from_json(step_params)
-            self.steps.append(step)
             name_step[step.name] = step
+            if not step.run_if_requested:
+                self.steps.append(step)
```

Comment:
I don't think this works.  runIfRequested: false means the step isn't in self.steps, and the code below just filters self.steps.

Reply:
self.steps is unused and just overwritten after this.  I think this if/append is not needed.